metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/StringFunctions.java

```diff
@@ -450,4 +454,40 @@ public Object apply(List<Object> strings) {
     }
   }
 
+  @Stellar( name = "PARSE_JSON_STRING"
+          , description = "Returns a JSON object for the specified JSON string"
+          , params = {
+            "str - the JSON String to convert, may be null"
+          }
+          , returns = "an Object containing the parsed JSON string"
+  )
+  public static class ParseJsonString extends BaseStellarFunction {
+
```

Comment:
So we are just returning whatever we get back from Jackson to the caller.  Meaning that they have to understand the content of the json in order to write the expression correctly.  This needs to be documented if it stays like this.

I am not sure I like that.  Can you give the use cases or circumstances that where you imagine this being used?  What are the bounds?

Reply:
So, the usecase is having an enrichment which stores a JSON blob in HBase (or some other store) that you want to marshall into a Java object to interact with via stellar.  Some people would prefer to have their data in a readable format rather than serialized via kryo (like we do in the profiler) or restricted to a certain format (like we do in enrichments).  This opens up those use-cases.

That JSON blob may be a map or a list or a primitive; the idea is to let jackson handle the parsing of it, so the bounds are what JSON can represent.  The knowledge of what is returned would be necessary for the caller, though, as you say.

Reply:
Updated README

Reply:
What do you guys think of having PARSE_TO_MAP, PARSE_TO_XXXX functions to have **some** type safety?

Reply:
I don't mind having them as convenience functions in addition to the generic one.

Reply:
I think it will make more sense to authors, and have less of a surprise factor.

- you need to understand your data
- use the right function
- get good error checking and verification